lib/bap/bap_self.ml

```diff
@@ -184,26 +197,40 @@ module Create() = struct
             Stream.unsubscribe Plugins.events subscription
           | _ -> () )
 
-
-    let bool = Arg.bool
-    let char = Arg.char
-    let int = Arg.int
-    let nativeint = Arg.nativeint
-    let int32 = Arg.int32
-    let int64 = Arg.int64
-    let float = Arg.float
-    let string = Arg.string
-    let enum = Arg.enum
-    let file = Arg.file
-    let dir = Arg.dir
-    let non_dir_file = Arg.non_dir_file
-    let list = Arg.list
-    let array = Arg.array
-    let pair = Arg.pair
-    let t2 = Arg.t2
-    let t3 = Arg.t3
-    let t4 = Arg.t4
-    let some = Arg.some
+    let converter : 'a Arg.converter -> 'a -> 'a converter =
+      fun (parser, printer) default -> parser, printer, default
+
+    let bool = converter Arg.bool false
+    let char = converter Arg.char '\x00'
```

Comment:
It is `x` that is arbitrary, since there is no specific logic why we chose them. If someone will see that it is zero, then he will try to use:

```ocaml
let var = Config.(param char "var")
...
let x = if !var = '\x00' then create_fresh () else var
...
```

That I would like to discourage in general, as one should use at least option for this. There is another concern, that if this value will slip into the man page, will it be rendered correctly. As it is non-printable special character.